tests/test_upload.py

```diff
@@ -38,19 +32,43 @@ def test_upload(copy_sample):
 """
 # That's not a real password. Well, hopefully not.
 
+@contextmanager
+def temp_pypirc(content):
+    try:
+        temp_file = NamedTemporaryFile("w+", delete=False)
+        temp_file.write(content)
+        temp_file.close()
+        yield temp_file.name
+    finally:
+        os.unlink(temp_file.name)
```

Comment:
Pytest has a handy shortcut for this - if you use a `tmp_path` fixture in a test, it creates a temporary directory, you can write whatever you want there, and it will all be cleaned up when the test runs. [Docs here](https://docs.pytest.org/en/latest/how-to/tmp_path.html#tmp-path)

I'm happy with this, the fixture is just a neat thing to know about. :slightly_smiling_face:

Reply:
Thanks! Good to learn about that :)